File: Go/player.go
package main

import "fmt"

type player struct {
	cards []*card
	name  string
	score byte
}

func (p *player) addCard(card *card) {
	p.score += card.value
	p.cards = append(p.cards, card)
}

func (p *player) display(mask bool) {
	if mask {
		fmt.Printf("%s %d? %s [**]\n", p.name, p.cards[0].value, p.cards[0].format())
	} else {
		cardStr := ""

		for _, card := range p.cards {
			cardStr += card.format() + " "
		}

		fmt.Printf("%s %d %s\n", p.name, p.score, cardStr)
	}
}

File: Rust/src/player.rs
use crate::card::Card;

#[derive(Debug)]
pub struct Player {
    point: u8,
    cards: Vec<Card>,
}

impl Player {
    pub fn init(card1: Card, card2: Card) -> Self {
        Self {
            point: card1.val() + card2.val(),
            cards: vec![card1, card2],
        }
    }

    pub fn point(&self) -> u8 {
        self.point
    }

    /** 手札追加 */
    pub fn add(&mut self, card: Card) {
        self.point += card.val();
        self.cards.push(card);
    }

    /** カードを表示用の文字列に */
    fn join_cards(&self) -> String {
        self.cards
            .iter()
            .map(|x| x.disp())
            .collect::<Vec<String>>()
            .join(" ")
    }

    /** 手札を表示 */
    pub fn show(&self, hidden: bool) -> String {
        if hidden {
            format!("** [{} **]", self.cards[0].disp())
        } else {
            format!("{} [{}]", self.point, self.join_cards())
        }
    }
}
